### ISSUE TITLE
how to add subSection ?

### ISSUE THREAD

**Issue by ystyle** [bmaupin/go-epub#10]
how to add subSection ?

**Comment by bmaupin** [comment]
I'm sorry it's taken me so long to get back to you.

Could you explain what you mean by "subSection?" I'm asking because it's not really an official term in the EPUB specification, so I can better help you if I know exactly what you're asking for. 

For example, this library has the ability to add "sections," which are saved as separate files in the resulting EPUB and added to the table of contents.

Thanks!

**Comment by ystyle** [comment]
like this [psiegman/epublib](https://github.com/psiegman/epublib)
```java
 // Add Chapter 2
      TOCReference chapter2 = book.addSection( "Second Chapter",
        getResource("/book1/chapter2.html", "chapter2.html") );
// Add Chapter2, Section 1
      book.addSection(chapter2, "Chapter 2, section 1",
        getResource("/book1/chapter2_1.html", "chapter2_1.html"));
```

![](https://dl.ystyle.top/images/2018-12/calibre-parallel_2018-12-29_14-30-23.png)

**Comment by bmaupin** [comment]
Ah, okay, so it sounds like you want the ability to add content within sections to the table of contents, right?

There currently isn't a way to do that, although pull requests are welcome!

**Comment by bmaupin** [comment]
I think I remember now why I didn't think to add this... For the eReaders I've used, they flatten every single TOC entry into a really long list and it becomes very annoying to scroll through. So typically when creating EPUBs I only add TOC entries for the major sections/chapters.

Of course that's not to say that it wouldn't be a nice feature to have. I don't have time to do this myself right now, so I've tagged this issue in case someone else wants to pick it up.

Thanks!

**Comment by ystyle** [comment]
I temporarily added this feature, but it may not be implemented well, but it can solve the problem.

Code is here, I don't know if it should be create a pr

https://github.com/ystyle/go-epub/commit/76f08f231a732c83302b084a3dfd85f26f08b92c

**Comment by bmaupin** [comment]
Oh cool, thanks! I'm working on a bug for another project but once I'm done with that I'll take a look.

**Comment by bmaupin** [comment]
Closed by #58
